jupyterlab_git/git.py

```diff
@@ -103,10 +107,17 @@ def config(self, top_repo_path, **kwargs):
             else:
                 raw = output.decode("utf-8").strip()
                 response["options"] = dict()
+                previous_key = None
                 for l in raw.split("\n"):
-                    k, v = l.split("=", maxsplit=1)
-                    if k in ALLOWED_OPTIONS:
-                        response["options"][k] = v
+                    if l.startswith('"') and previous_key in ALLOWED_OPTIONS:
+                        response["options"][previous_key] += l
+                    elif "=" in l:
+                        key, v = l.split("=", maxsplit=1)
+                        previous_key = key
+                        if key in ALLOWED_OPTIONS:
+                            response["options"][key] = v
+                    else:
+                        logger.debug("[jupyterlab-git] Unable to interpret git option {}.".format(l))
```

Comment:
The parsing loop looks good for the most part. It looks like `"` is indeed the line continuation marker. However, there's at least one case I spotted from https://github.com/jupyterlab/jupyterlab-git/issues/498#issuecomment-564735865 that will fall through the parser here:

```
alias.summary=!f() {     printf "Summary of this branch...
";     printf "%s
" $(git rev-parse --abbrev-ref HEAD);     printf "%s first commit timestamp
" $(git log --date-order --format=%cI | tail -1);     printf "%s latest commit timestamp
" $(git log -1 --date-order --format=%cI);     printf "%d commit count
" $(git rev-list --count HEAD);     printf "%d date count
" $(git log --format=oneline --format="%ad" --date=format:"%Y-%m-%d" | awk '{a[$0]=1}END{for(i in a){n++;} print n}');     printf "%d tag count
" $(git tag | wc -l);     printf "%d author count
" $(git log --format=oneline --format="%aE" | awk '{a[$0]=1}END{for(i in a){n++;} print n}');     printf "%d committer count
" $(git log --format=oneline --format="%cE" | awk '{a[$0]=1}END{for(i in a){n++;} print n}');     printf "%d local branch count
" $(git branch | grep -v " -> " | wc -l);     printf "%d remote branch count
" $(git branch -r | grep -v " -> " | wc -l);     printf "
Summary of this directory...
";     printf "%s
" $(pwd);     printf "%d file count via git ls-files
" $(git ls-files | wc -l);     printf "%d file count via find command
" $(find . | wc -l);     printf "%d disk usage
" $(du -s | awk '{print $1}');     printf "
Most-active authors, with commit count and %%...
"; git log-of-count-and-email | head -7;     printf "
Most-active dates, with commit count and %%...
"; git log-of-count-and-day | head -7;     printf "
Most-active files, with churn count
"; git churn | head -7;   }; f
```

Reply:
Think you can figure out how to handle this case as well?

Reply:
Thanks a lot for the nice regex @telamonian. I updated the PR accordingly.